src/agents/players/book_agent.py

```diff
@@ -0,0 +1,218 @@
+    10  H  H  H  H  H  H  H  H  H  H
+    11  H  H  H  H  H  H  H  H  H  H
+    12  H  S  S  S  S  S  H  H  H  H
+    13  H  S  S  S  S  S  H  H  H  H
+    14  H  S  S  S  S  S  H  H  H  H
+    15  H  S  S  S  S  S  H  H  H  H
+    16  H  S  S  S  S  S  H  H  H  H
+    17  S  S  S  S  S  S  S  S  S  S
+    18  S  S  S  S  S  S  S  S  S  S
+    19  S  S  S  S  S  S  S  S  S  S
+    20  S  S  S  S  S  S  S  S  S  S
+    21  S  S  S  S  S  S  S  S  S  S
+    """
+    hard_actions_ndx_offset = 6
+    hard_actions = [
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, stay, stay, stay, stay, stay, hit, hit, hit, hit],
+        [hit, stay, stay, stay, stay, stay, hit, hit, hit, hit],
+        [hit, stay, stay, stay, stay, stay, hit, hit, hit, hit],
+        [hit, stay, stay, stay, stay, stay, hit, hit, hit, hit],
+        [hit, stay, stay, stay, stay, stay, hit, hit, hit, hit],
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay],
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay],
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay],
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay],
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay]
+    ]
+    
+    
+    """
+    by the book action matrix for hands that aren't soft but we can split
+        1  2  3  4  5  6  7  8  9  10  
+    4   H  H  H  H  H  H  H  H  H  H
+    6   H  H  H  H  H  H  H  H  H  H
+    8   H  H  H  H  H  H  H  H  H  H
+    10  H  D  D  D  D  H  H  H  H  H
+    12  H  H  H  H  H  H  H  H  H  H
+    14  P  P  P  P  P  P  P  P  P  P
+    16  P  P  P  P  P  P  P  P  P  P
+    18  P  P  P  P  P  P  P  P  P  P
+    20  S  S  S  S  S  S  S  S  S  S
+    """
+    split_action_ndx_offset = 4
+    split_action_ndx_multiplier = 2
+    split_actions = [
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [hit, double, double, double, double, hit, hit, hit, hit, hit],
+        [hit, hit, hit, hit, hit, hit, hit, hit, hit, hit],
+        [split, split, split, split, split, split, split, split, split, split],
+        [split, split, split, split, split, split, split, split, split, split],
+        [split, split, split, split, split, split, split, split, split, split],
+        [stay, stay, stay, stay, stay, stay, stay, stay, stay, stay]
+    ]
+
+    def __init__(self):
+        super().__init__()
+
+        self.dealer_card = 0
+
+    def update_model(self, game):
+        self.dealer_card = game.dealer.hands[0].cards[1]
```

Comment:
I think this should be `cards[0]`

Reply:
is it the first or second card that is dealt that the player sees?

Reply:
First